src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java

```diff
@@ -426,6 +427,44 @@ public void startSearcherPruningThread(CountDownLatch shutdownNow) {
     }
   }
 
+  /**
+   * Factory class that produces a new searcher for each {@link IndexReader} version. Sets field
+   * similarity and handles any eager global ordinal loading.
+   */
+  private class ShardSearcherFactory extends SearcherFactory {
+    private final boolean loadEagerOrdinals;
+
+    /**
+     * Constructor.
+     *
+     * @param loadEagerOrdinals is eager global ordinal loading enabled, not needed for primary
+     */
+    ShardSearcherFactory(boolean loadEagerOrdinals) {
+      this.loadEagerOrdinals = loadEagerOrdinals;
+    }
+
+    @Override
+    public IndexSearcher newSearcher(IndexReader reader, IndexReader previousReader)
+        throws IOException {
+      IndexSearcher searcher = new MyIndexSearcher(reader, searchExecutor);
+      searcher.setSimilarity(indexState.sim);
+      if (loadEagerOrdinals) {
+        loadEagerGlobalOrdinals(reader);
+      }
+      return searcher;
+    }
+
+    private void loadEagerGlobalOrdinals(IndexReader reader) throws IOException {
+      for (Map.Entry<String, FieldDef> entry : indexState.eagerGlobalOrdinalFields.entrySet()) {
+        // only sorted set doc values facet currently supported
+        if (entry.getValue().getFacetValueType() == FacetValueType.SORTED_SET_DOC_VALUES) {
```

Comment:
If `eagerGlobalOrdinals` flag is set for a field that does not have `SORTED_SET_DOC_VALUES`, should we check and disallow that at mapping creation time or at least log a warning in here?